doc/src/bond_cpp.md

```diff
@@ -185,6 +185,59 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Type evolution
+==============
+
+Bond does not use explicit versioning to deal with changes to schemas over time
+Instead, Bond supports certain type evolution operations which allow the
+producer and consumer of Bond types to evolve without lockstep coordination.
+
+The following changes to a type will never break compatibility across the wire:
+
+- Adding or removing an `optional` or `optional_required` field
+- Changing type between `int32` and `enum`
+- Changing type between `vector<T>` and `list<T>`
+- Changing type between `blob` and `vector<uint8>` or `blob` and `list<uint8>`
+- Changing type between `T` and `bonded<T>`
+- Adding new enumeration values at the end of an enumeration
```

Comment:
You _can_ add an enumeration value in the middle, so long as it doesn't change the value of anything else. This is safe:

    enum Foo { Bar = 3; Baz = 2; }
    enum Foo { Bar = 3; Qux = 7; Baz = 2; }

"Adding new enumeration contants without perturbing the numeric value of the existing constants."?

We use "constant" to refer to the things inside the enum in the compiler documentation. Though, I often refer to them as enum member of enumerated values. Alas, enumerant isn't a word.